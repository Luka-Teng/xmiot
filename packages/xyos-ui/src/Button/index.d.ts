import * as React from 'react'

type ButtonType = 'default' | 'primary' | 'ghost' | 'dashed' | 'danger'
type ButtonHtmlType = 'submit' | 'button' | 'reset'
type ButtonSize = 'medium' | 'large' | 'small'

export interface ButtonProps {
  type?: ButtonType;
  icon?: string;
  size?: ButtonSize;
  loading?: boolean | { delay?: number };
  disabled?: boolean
  className?: string;
  block?: boolean;
  htmlType?: ButtonHtmlType
  children?: React.ReactNode;
  onClick?: React.MouseEventHandler<HTMLElement>;
  ghost?:boolean
}


declare class Button extends React.Component<ButtonProps> {}

export default Button
