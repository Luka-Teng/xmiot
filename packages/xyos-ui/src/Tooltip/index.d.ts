import React from 'react'
import { TriggerProps } from 'rc-trigger'
import {
  AlignType,
  AnimationType,
  ActionType,
  BuildInPlacements
} from 'rc-trigger/lib/interface'

export type TooltipPlacement =
  | 'top'
  | 'left'
  | 'right'
  | 'bottom'
  | 'topLeft'
  | 'topRight'
  | 'bottomLeft'
  | 'bottomRight'
  | 'leftTop'
  | 'leftBottom'
  | 'rightTop'
  | 'rightBottom'

export interface AdjustOverflow {
  adjustX?: 0 | 1
  adjustY?: 0 | 1
}

export type RenderFunction = () => React.ReactNode

export interface AbstractTooltipProps {
  prefixCls?: string
  overlayClassName?: string
  style?: React.CSSProperties
  className?: string
  overlayStyle?: React.CSSProperties
  placement?: TooltipPlacement
  builtinPlacements?: BuildInPlacements
  defaultVisible?: boolean
  visible?: boolean
  onVisibleChange?: (visible: boolean) => void
  mouseEnterDelay?: number
  mouseLeaveDelay?: number
  transitionName?: string
  trigger?: ActionType
  openClassName?: string
  arrowPointAtCenter?: boolean
  autoAdjustOverflow?: boolean | AdjustOverflow
  // getTooltipContainer had been rename to getPopupContainer
  getTooltipContainer?: (triggerNode: HTMLElement) => HTMLElement
  getPopupContainer?: (triggerNode: HTMLElement) => HTMLElement
  children?: React.ReactNode
  // align is a more higher api
  align?: AlignType
  /** Internal. Hide tooltip when hidden. This will be renamed in future. */
  destroyTooltipOnHide?: boolean
  content?: React.ReactNode | RenderFunction
}
export interface TooltipPropsWithOverlay extends AbstractTooltipProps {
  title?: React.ReactNode | RenderFunction
  overlay: React.ReactNode | RenderFunction
}

export interface TooltipPropsWithTitle extends AbstractTooltipProps {
  title: React.ReactNode | RenderFunction
  overlay?: React.ReactNode | RenderFunction
}

export declare type TooltipProps =
  | TooltipPropsWithTitle
  | TooltipPropsWithOverlay

declare class Tooltip extends React.Component<TooltipProps> {}
export default Tooltip
