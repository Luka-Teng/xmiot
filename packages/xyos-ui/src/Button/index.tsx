import './lib/button.less'

export { default as Button } from './lib/Button'


