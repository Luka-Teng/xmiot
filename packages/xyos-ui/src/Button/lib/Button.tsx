import React from 'react'
import classnames from 'classnames'

export const tuple = <T extends string[]>(...args: T) => args;

export type ButtonType = 'default' | 'primary' | 'ghost' | 'dashed' | 'danger'
type ButtonHtmlType = 'submit' | 'button' | 'reset'
type ButtonSize = 'medium' | 'large' | 'small'

export interface ButtonProps {
  type?: ButtonType;
  icon?: string;
  size?: ButtonSize;
  loading?: boolean | { delay?: number };
  disabled?: boolean
  className?: string;
  block?: boolean;
  htmlType?: ButtonHtmlType
  children?: React.ReactNode;
  onClick?: React.MouseEventHandler<HTMLElement>;

}

interface ButtonState {
  loading?: boolean | { delay?: number };
}

class Button extends React.Component<ButtonProps, ButtonState> {

  static defaultProps = {
    type: 'default',
    loading: false,
    ghost: false,
    block: false,
    htmlType: 'button',
  };

  constructor(props: ButtonProps) {
    super(props)
    this.state = {
      loading: props.loading,
    };
  }

  render() {
    const { children } = this.props
    const { className, type, htmlType, size, disabled, loading, icon } = this.props
    const _className = classnames(['button', 'button-' + type, 'button-' + size, className], {
      ['button-disabled']: disabled,
      [`button-loading`]: !!loading,
    })
    const iconNode = loading ? 'loading' : icon;

    return (
      <button className={_className} onClick={this.onClick} type={htmlType}>
        {
          loading && <img className='action-loading' src={require('./loading.gif')} alt="" />
        }
        <span>{children}</span>
      </button>
    )
  }

  private onClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    const { onClick, disabled, loading } = this.props
    if (onClick) {
      if (!disabled && !loading) {
        onClick(event)
      }
    }
  }

}

export default Button

