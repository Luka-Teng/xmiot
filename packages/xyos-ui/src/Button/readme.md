## Button

|props| description                 |
|:----|:------------------|
|type | button 类型 ('default'（默认）,primary,ghost,dashed,danger'）
|size | button 大小 （normal，large ，small）|
|loading | button loading的状态 （boolean）|
|disabled| button 不可用的状态 （boolean）|

----

```
 <Button type="primary">
   Primary
 </Button>

  <Button type="primary" disabled>
    Primary(disabled)
  </Button>

  <Button type="primary" loading>
    加载
  </Button>

  <Button type="primary" size='small'>
    small(primary)
  </Button>

  <Button  size='small'>
    small
  </Button>
  
  <Button type="primary" size='large'>
    lg(primary)
  </Button>
```
