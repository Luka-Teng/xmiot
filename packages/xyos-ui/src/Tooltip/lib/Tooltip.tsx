import * as React from 'react'
import RcTooltip from 'rc-tooltip'
import classnames from 'classnames'
import {
  AlignType,
  ActionType,
  BuildInPlacements
} from 'rc-trigger/lib/interface'
import getPlacements, { AdjustOverflow, PlacementsConfig } from './placements'
import './tooltip.less'

// export { AdjustOverflow, PlacementsConfig };

export interface ConfigConsumerProps {
  getPopupContainer?: (triggerNode: HTMLElement) => HTMLElement
  rootPrefixCls?: string
  getPrefixCls: (suffixCls: string, customizePrefixCls?: string) => string
  pageHeader?: {
    ghost: boolean
  }
  direction?: 'ltr' | 'rtl'
}

export type TooltipPlacement =
  | 'top'
  | 'left'
  | 'right'
  | 'bottom'
  | 'topLeft'
  | 'topRight'
  | 'bottomLeft'
  | 'bottomRight'
  | 'leftTop'
  | 'leftBottom'
  | 'rightTop'
  | 'rightBottom'

export type RenderFunction = () => React.ReactNode

export interface AbstractTooltipProps {
  prefixCls?: string
  overlayClassName?: string
  style?: React.CSSProperties
  className?: string
  overlayStyle?: React.CSSProperties
  placement?: TooltipPlacement
  builtinPlacements?: BuildInPlacements
  defaultVisible?: boolean
  visible?: boolean
  onVisibleChange?: (visible: boolean) => void
  mouseEnterDelay?: number
  mouseLeaveDelay?: number
  transitionName?: string
  trigger?: ActionType
  openClassName?: string
  arrowPointAtCenter?: boolean
  autoAdjustOverflow?: boolean | AdjustOverflow
  // getTooltipContainer had been rename to getPopupContainer
  getTooltipContainer?: (triggerNode: HTMLElement) => HTMLElement
  getPopupContainer?: (triggerNode: HTMLElement) => HTMLElement
  children?: React.ReactNode
  // align is a more higher api
  align?: AlignType
  /** Internal. Hide tooltip when hidden. This will be renamed in future. */
  destroyTooltipOnHide?: boolean
  content?: React.ReactNode | RenderFunction
}
export interface TooltipPropsWithOverlay extends AbstractTooltipProps {
  title?: React.ReactNode | RenderFunction
  overlay: React.ReactNode | RenderFunction
}

export interface TooltipPropsWithTitle extends AbstractTooltipProps {
  title: React.ReactNode | RenderFunction
  overlay?: React.ReactNode | RenderFunction
}

export interface PopoverProps extends AbstractTooltipProps {
  title?: React.ReactNode | RenderFunction

  overlay?: React.ReactNode | RenderFunction
}

export declare type TooltipProps =
  | TooltipPropsWithTitle
  | TooltipPropsWithOverlay

interface TooltipState {
  visible: boolean
}

const splitObject = (obj: any, keys: string[]) => {
  const picked: any = {}
  const omitted: any = { ...obj }
  keys.forEach(key => {
    if (obj && key in obj) {
      picked[key] = obj[key]
      delete omitted[key]
    }
  })
  return { picked, omitted }
}

function getDisabledCompatibleChildren (element: React.ReactElement<any>) {
  const elementType = element.type as any
  if (
    (elementType.__XYOS_BUTTON === true ||
      elementType.__ANT_SWITCH === true ||
      elementType.__ANT_CHECKBOX === true ||
      element.type === 'button') &&
    element.props.disabled
  ) {
    // Pick some layout related style properties up to span
    const { picked, omitted } = splitObject(element.props.style, [
      'position',
      'left',
      'right',
      'top',
      'bottom',
      'float',
      'display',
      'zIndex'
    ])
    const spanStyle = {
      display: 'inline-block', // default inline-block is important
      ...picked,
      cursor: 'not-allowed',
      width: element.props.block ? '100%' : null
    }
    const buttonStyle = {
      ...omitted,
      pointerEvents: 'none'
    }
    const child = React.cloneElement(element, {
      style: buttonStyle,
      className: null
    })
    return (
      <span style={spanStyle} className={element.props.className}>
        {child}
      </span>
    )
  }
  return element
}
class Tooltip extends React.Component<TooltipProps, TooltipState> {
  static defaultProps = {
    placement: 'top' as TooltipPlacement,
    // transitionName: 'zoom-big-fast',
    mouseEnterDelay: 0.1,
    mouseLeaveDelay: 0.1,
    arrowPointAtCenter: false,
    autoAdjustOverflow: true,
    target: 'hover'
  }

  static getDerivedStateFromProps (nextProps: TooltipProps) {
    if ('visible' in nextProps) {
      return { visible: nextProps.visible }
    }
    return null
  }

  private tooltip = React.createRef<typeof RcTooltip>()

  constructor (props: TooltipProps) {
    super(props)
    this.state = {
      visible: !!props.visible || !!props.defaultVisible
    }
  }

  isNoTitle () {
    const { title, overlay } = this.props
    return !title && !overlay // overlay for old version compatibility
  }

  getPopupDomNode () {
    return (
      this.tooltip.current && (this.tooltip.current as any).getPopupDomNode()
    )
  }

  // 显示隐藏的回调
  onVisibleChange = (visible: boolean) => {
    const { onVisibleChange } = this.props
    if (!('visible' in this.props)) {
      this.setState({ visible: this.isNoTitle() ? false : visible })
    }
    if (onVisibleChange && !this.isNoTitle()) {
      onVisibleChange(visible)
    }
  }

  getPlacements () {
    const {
      builtinPlacements,
      arrowPointAtCenter,
      autoAdjustOverflow
    } = this.props
    return (
      builtinPlacements ||
      getPlacements({
        arrowPointAtCenter,
        verticalArrowShift: 8,
        autoAdjustOverflow
      })
    )
  }

  // 动态设置动画点
  onPopupAlign = (domNode: HTMLElement, align: any) => {
    const placements: any = this.getPlacements()

    // 当前返回的位置
    const placement = Object.keys(placements).filter(
      key =>
        placements[key].points[0] === align.points[0] &&
        placements[key].points[1] === align.points[1]
    )[0]
    if (!placement) {
      return
    }
    // 根据当前坐标设置动画点
    const rect = domNode.getBoundingClientRect()
    const transformOrigin = {
      top: '50%',
      left: '50%'
    }
    if (placement.indexOf('top') >= 0 || placement.indexOf('Bottom') >= 0) {
      transformOrigin.top = `${rect.height - align.offset[1]}px`
    } else if (
      placement.indexOf('Top') >= 0 ||
      placement.indexOf('bottom') >= 0
    ) {
      transformOrigin.top = `${-align.offset[1]}px`
    }
    if (placement.indexOf('left') >= 0 || placement.indexOf('Right') >= 0) {
      transformOrigin.left = `${rect.width - align.offset[0]}px`
    } else if (
      placement.indexOf('right') >= 0 ||
      placement.indexOf('Left') >= 0
    ) {
      transformOrigin.left = `${-align.offset[0]}px`
    }
    domNode.style.transformOrigin = `${transformOrigin.left} ${
      transformOrigin.top
    }`
  }

  getOverlay () {
    const prefixCls = 'xyos-tooltip'
    const { title, content } = this.props
    if (title && !content) {
      return title
    }
    if (title && content) {
      return (
        <div>
          {title && <div className={`${prefixCls}-popover-title`}>{title}</div>}
          <div className={`${prefixCls}-popover-inner-content`}>{content}</div>
        </div>
      )
    }
  }

  render () {
    const { props, state } = this
    const {
      title,
      content,
      overlay,
      openClassName,
      getPopupContainer,
      getTooltipContainer,
      overlayClassName
    } = props

    let { visible } = state

    let direction

    const children = props.children as React.ReactElement<any>

    const child = getDisabledCompatibleChildren(
      React.isValidElement(children) ? children : <span>{children}</span>
    )

    const childProps = child.props

    const prefixCls = 'xyos-tooltip'

    const childCls = classnames(childProps.className, {
      [openClassName || `${prefixCls}-open`]: true
    })

    const customOverlayClassName = classnames(overlayClassName, {
      [`${prefixCls}-rtl`]: direction === 'rtl',
      [`${prefixCls}-notitle`]: !content
    })

    return (
      <RcTooltip
        {...this.props}
        prefixCls={prefixCls}
        getTooltipContainer={getPopupContainer || getTooltipContainer}
        ref={this.tooltip}
        overlayClassName={customOverlayClassName}
        overlay={this.getOverlay()}
        builtinPlacements={this.getPlacements()}
        onVisibleChange={this.onVisibleChange}
        visible={visible}
        onPopupAlign={this.onPopupAlign}
      >
        {visible ? React.cloneElement(child, { className: childCls }) : child}
      </RcTooltip>
    )
  }
}

export default Tooltip
