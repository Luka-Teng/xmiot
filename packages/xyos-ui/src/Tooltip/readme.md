## Tooltip

|props         | description              | 类型     |
|:-------------|:-------------------------|:---------|
|title         |  提示文字                | string|ReactNode|() => ReactNode
|arrowPointAtCenter | 箭头是否指向目标元素中心   | boolean (false)
|autoAdjustOverflow | 气泡被遮盖住 自动调整位置 | boolean （true）
|defaultVisible | 是否默认隐藏 | boolean（false）
|getPopupContainer |	浮层渲染父节点，默认渲染到 body 上 |Function(triggerNode)
|mouseEnterDelay | 	鼠标移入后延时多少才显示 Tooltip，单位：秒 |number
| mouseLeaveDelay | 鼠标移出后延时多少才隐藏 Tooltip，单位：秒 | number
|overlayClassName |卡片类名 | string
|overlayStyle | 卡片样式 | object
| placement | 气泡框位置，可选 top left right bottom topLeft topRight bottomLeft bottomRight leftTop leftBottom rightTop rightBottom | string
|trigger  |触发行为，可选 hover/focus/click/ | string
|visible  | 用于手动控制浮层 | boolean
| onVisibleChange | 显示隐藏的回调 | 	(visible) => void
|title         |  卡片内容                | string|ReactNode|() => ReactNode
------




