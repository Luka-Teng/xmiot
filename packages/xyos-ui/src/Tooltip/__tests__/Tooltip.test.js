import React from 'react'
import { mount } from 'enzyme'
import Tooltip from '../../Tooltip'
import Button from '../../Button'

// eslint-disable-next-line jest/no-export
function mountTest(Component) {
  describe(`mount and unmount`, () => {
    it(`component could be updated and unmounted without errors`, () => {
      const wrapper = mount(<Component />);
      expect(() => {
        wrapper.setProps({});
        wrapper.unmount();
      }).not.toThrow();
    });
  });
}


describe('test tooltip',()=>{
  mountTest(Tooltip);

  it('未触发的时候，是否挂载',()=>{
   const wrapper = mount(
    <Tooltip title='这是一个Tooltip，这是一个Tooltip'>
     <Button>TL</Button>
   </Tooltip>
   )
   expect(wrapper.find('.xyos-tooltip')).toMatchSnapshot()
  })


  it( 'check `onVisibleChange` arguments',()=>{
    const onVisibleChange = jest.fn();
    const wrapper = mount(
      <Tooltip title="" mouseEnterDelay={0} mouseLeaveDelay={0} onVisibleChange={onVisibleChange}>
        <div id="hello">Hello world!</div>
      </Tooltip>,
    );

    // `title` is empty.
    const div = wrapper.find('#hello').at(0);
    div.simulate('mouseenter');
    expect(onVisibleChange).not.toHaveBeenCalled();
    // console.log(wrapper.instance().tooltip.current.props,'000')
    expect(wrapper.instance().tooltip.current.props.visible).toBe(false);

    div.simulate('mouseleave');
    expect(onVisibleChange).not.toHaveBeenCalled();
    expect(wrapper.instance().tooltip.current.props.visible).toBe(false);

    // update `title` value.
    wrapper.setProps({ title: 'Have a nice day!' });
    wrapper.find('#hello').simulate('mouseenter');
    expect(onVisibleChange).toHaveBeenLastCalledWith(true);
    expect(wrapper.instance().tooltip.current.props.visible).toBe(true);

    // add `visible` props.
    wrapper.setProps({ visible: false });
    wrapper.find('#hello').simulate('mouseenter');
    expect(onVisibleChange).toHaveBeenLastCalledWith(true);
    const lastCount = onVisibleChange.mock.calls.length;
    expect(wrapper.instance().tooltip.current.props.visible).toBe(false);

    // always trigger onVisibleChange
    wrapper.simulate('mouseleave');
    expect(onVisibleChange.mock.calls.length).toBe(lastCount); // no change with lastCount
    expect(wrapper.instance().tooltip.current.props.visible).toBe(false);
  })

  it('should hide when mouse leave native disabled button', () => {
    const onVisibleChange = jest.fn();
    const wrapper = mount(
      <Tooltip
        title="xxxxx"
        mouseEnterDelay={0}
        mouseLeaveDelay={0}
        onVisibleChange={onVisibleChange}
      >
        <button type="button" disabled>
          Hello world!
        </button>
      </Tooltip>,
    );

    expect(wrapper.find('span')).toHaveLength(1);
    const button = wrapper.find('span').at(0);
    button.simulate('mouseenter');
    expect(onVisibleChange).toHaveBeenCalledWith(true);
    expect(wrapper.instance().tooltip.current.props.visible).toBe(true);

    button.simulate('mouseleave');
    expect(onVisibleChange).toHaveBeenCalledWith(false);
    expect(wrapper.instance().tooltip.current.props.visible).toBe(false);
  });

  describe('should hide when mouse leave antd disabled component', () => {
    function testComponent(name, Component) {
      it(name, () => {
        const onVisibleChange = jest.fn();
        const wrapper = mount(
          <Tooltip
            title="xxxxx"
            mouseEnterDelay={0}
            mouseLeaveDelay={0}
            onVisibleChange={onVisibleChange}
          >
            <Component disabled />
          </Tooltip>,
        );

        expect(wrapper.render()).toMatchSnapshot();
        const button = wrapper.find('span').at(0);
        button.simulate('mouseenter');
        expect(onVisibleChange).toHaveBeenCalledWith(true);
        expect(wrapper.instance().tooltip.current.props.visible).toBe(true);

        button.simulate('mouseleave');
        expect(onVisibleChange).toHaveBeenCalledWith(false);
        expect(wrapper.instance().tooltip.current.props.visible).toBe(false);
      });
    }

    testComponent('Button', Button);
  });

  it('use arrowPointAtCenter', () => {
    const wrapper = mount(
      <div>
        <Tooltip placement="topLeft" title="Prompt Text">
          <Button>Align edge / 边缘对齐</Button>
        </Tooltip>
        <Tooltip placement="topLeft" title="Prompt Text" arrowPointAtCenter>
          <Button>Arrow points to center / 箭头指向中心</Button>
        </Tooltip>
      </div>
    );
    expect(wrapper.find('.xyos-tooltip')).toMatchSnapshot()
  })

  it('use placement',() => {
    const text='测试'
    const wrapper = mount(<div>
          <div style={{ marginLeft: 60 }}>
            <Tooltip placement="topLeft" title={text}>
              <a href="#">TL</a>
            </Tooltip>
            <Tooltip placement="top" title={text}>
              <a href="#">Top</a>
            </Tooltip>
            <Tooltip placement="topRight" title={text}>
              <a href="#">TR</a>
            </Tooltip>
          </div>
          <div style={{ width: 60, float: 'left' }}>
            <Tooltip placement="leftTop" title={text}>
              <a href="#">LT</a>
            </Tooltip>
            <Tooltip placement="left" title={text}>
              <a href="#">Left</a>
            </Tooltip>
            <Tooltip placement="leftBottom" title={text}>
              <a href="#">LB</a>
            </Tooltip>
          </div>
          <div style={{ width: 60, marginLeft: 270 }}>
            <Tooltip placement="rightTop" title={text}>
              <a href="#">RT</a>
            </Tooltip>
            <Tooltip placement="right" title={text}>
              <a href="#">Right</a>
            </Tooltip>
            <Tooltip placement="rightBottom" title={text}>
              <a href="#">RB</a>
            </Tooltip>
          </div>
          <div style={{ marginLeft: 60, clear: 'both' }}>
            <Tooltip placement="bottomLeft" title={text}>
              <a href="#">BL</a>
            </Tooltip>
            <Tooltip placement="bottom" title={text}>
              <a href="#">Bottom</a>
            </Tooltip>
            <Tooltip placement="bottomRight" title={text}>
              <a href="#">BR</a>
            </Tooltip>
          </div>
        </div>)
     expect(wrapper.find('.xyos-tooltip')).toMatchSnapshot()
  })
})