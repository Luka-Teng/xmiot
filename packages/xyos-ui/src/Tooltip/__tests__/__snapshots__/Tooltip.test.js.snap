// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test tooltip should hide when mouse leave antd disabled component Button 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "display: inline-block; cursor: not-allowed;",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "button button-default button-undefined button-disabled",
          "style": "pointer-events: none;",
          "type": "button",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
          "type": undefined,
        },
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`test tooltip use arrowPointAtCenter 1`] = `ReactWrapper {}`;

exports[`test tooltip use placement 1`] = `ReactWrapper {}`;

exports[`test tooltip 未触发的时候，是否挂载 1`] = `ReactWrapper {}`;
